AUTHENTICATION.md

```diff
@@ -0,0 +1,24 @@
+## How can assumeRoleARN be used with provider-jet-aws ?
```

Comment:
Would it make sense to just redirect people to https://github.com/crossplane/provider-aws/blob/master/AUTHENTICATION.md ? I think we fully cover all methods now, right?

Reply:
Yes could be also one option externalID is missing since we will add this with v0.24 ;) but then yes

Reply:
@muvaf so i can remove the md and add a link in readme ?

Reply:
Just to avoid confusion, I think it's better to have this doc in this repo instead of redirecting the users to different repo.

Reply:
@vara-bonthu Good point. We can possibly copy the doc from there and change all provider-aws references to provider-jet-aws because it's more extensive and covers more cases.